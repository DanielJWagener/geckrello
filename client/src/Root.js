// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
