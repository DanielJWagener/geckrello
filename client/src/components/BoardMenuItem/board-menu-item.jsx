import React from "react";

import "./board-menu-item.styles.scss";

const BoardMenuItem = ({ togglePanel, toggleTarget, children }) => (
  <button className="boardmenu__item" onClick={() => togglePanel(toggleTarget)}>
    {children}
  </button>
);

export default BoardMenuItem;
