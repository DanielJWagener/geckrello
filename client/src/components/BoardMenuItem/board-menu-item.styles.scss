@import "../../sass/abstracts/_variables.scss";

.boardmenu__item {
  font-size: 2rem;
  padding: 2rem 1rem 2rem 1rem;
  width: 100%;
  text-align: left;
  border: none;
  border-radius: 3px;
  background-color: $component-background;
  cursor: pointer;
  &:hover {
    background-color: $component-background-dark;
  }
}
