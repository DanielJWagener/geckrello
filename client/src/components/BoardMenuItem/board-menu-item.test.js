import { shallow } from "enzyme";
import React from "react";
import BoardMenuItem from "./board-menu-item.component";
import Root from "../../Root";

it("renders BoardMenuItem component", () => {
  expect(shallow(<BoardMenuItem />).debug()).toMatchSnapshot();
});
