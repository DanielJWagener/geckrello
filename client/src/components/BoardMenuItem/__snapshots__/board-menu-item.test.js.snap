// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardMenuItem component 1`] = `ShallowWrapper {}`;
